_internals/software-architecture.md

```diff
@@ -5,77 +5,57 @@ nav_order: 1
 use_mermaid: true
 ---
 
-Babelfish implements Microsoft SQL Server behavior on top of PostgreSQL. To achieve this goal, a
-couple of areas had to be modified inside the PostgreSQL core. PostgreSQL does
-not offer this capability out-of-the-box, therefore add-ons had to be provided
-to make this happen.
+Babelfish uses [hooks](/docs/internals/postgresql-hooks) to implement Microsoft SQL Server behavior
+from a PostgreSQL server. This allows the PostgreSQL server to speak more than one language, making
+the setup flexible and easy to manage.
 
-The way Babelfish implements Microsoft SQL Server behavior is by using
-[hooks](/docs/internals/postgresql-hooks). The idea is to make database protocols &ldquo;pluggable&rdquo;,
-which means that PostgreSQL can present itself as a different database. In fact, it
-is even possible to speak more than a single protocol at the same time, making
-the setup more flexible and easy to handle.
-
-<img src="/assets/images/babel_architecture.png" title="System architecture" width="800"/>
-
-Once the TDS handler has accepted the request coming in on the default port 1433,
-and once it has processed it, it is passed on via pltds to a standard PostgreSQL backend.
-PostgreSQL will still be exposed to the client on port 5432 (default).
-
-To sum it up: default:
+By default, Babelfish listens for connections on two TCP ports, in two dialects:
 
 - SQL Server dialect, clients connect to port 1433.
 - PostgreSQL dialect, clients connect to port 5432.
 
-In other words, a Babelfish enabled database will listen for connections on two
-TCP ports instead of one. Administrators have to take that into account. The ports
-used can be configured using PostgreSQL parameters.
 
-How does that work? The TDS listener captures the requests coming
-from the client and sends them though a parser for the SQL dialect of Microsoft
-SQL Server.  The query will then be handled by a regular PostgreSQL
-backend. The optimizer and the executor will basically be the same.
+<img src="/assets/images/babel_architecture.png" title="System architecture" width="800"/>
+
+Once the TDS handler has accepted and processed a request coming in on port 1433,
```

Comment:
It'd be better to not refer to 1433, since users can change that. I'm thinking "coming in on the TDS port", and then tweak line 14 above.

Reply:
I moved this line to above the graphic - I think it fits better with the content up there...

Reply:
All references to 1433 are removed in the last commit, and the port is noted as 'configurable'....